railties/lib/rails/application/configuration.rb

```diff
@@ -34,6 +34,7 @@ def initialize(*)
         @filter_redirect                         = []
         @helpers_paths                           = []
         @hosts                                   = Array(([".localhost", IPAddr.new("0.0.0.0/0"), IPAddr.new("::/0")] if Rails.env.development?))
+        @host_authorization                      = { response_app: action_dispatch.hosts_response_app }
```

Comment:
This is referencing `action_dispatch.hosts_response_app` too early, which will prevent applications from being able to set it. I think we should continue to pass it to the middleware, but emit a deprecation warning if it's set.

Reply:
I tested this an it appeared to work. I thought that `action_dispatch` would have been created prior to this. Is there a better way to pass the existing setting?

Reply:
I added a deprecation warning and a fallback to using `action_dispatch.hosts_response_app` if `response_app` is not supplied in the configuration. Is there anything else that needs to be done for deprecating a configuration setting? Let me know what you think of this solution.